### PR TITLE
Correct incorrect information in comparison

### PR DESCRIPTION
**Pull request by trickeydan**

### CODE REVIEW

File: docs/concepts/comparison.rst [hunk 1]
```diff
@@ -9,8 +9,6 @@ Similar Libraries
 +-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 |   Feature                                 | j5                       | robot-api / robotd                   | sr.robot                   |
 +===========================================+==========================+======================================+============================+
-| Compatible with sr.robot                  | Yes                      | No                                   | N / A                      |
-+-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 | Cross-Platform Support                    | Yes                      | No (Requires Linux + udev + systemd) | No (Requires Linux + udev) |
 +-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 | Custom  / Game Logic without core changes | Yes                      | No                                   | No                         |
```
Comment (RealOrangeOne) [inline docs/concepts/comparison.rst]: This line is moot for `j5`
```suggestion
| Custom  / Game Logic without core changes | N/A                      | No                                   | No                         |
```

File: docs/concepts/comparison.rst [hunk 2]
```diff
@@ -33,19 +31,17 @@ Similar Libraries
 +-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 | Supports multiple environments / backends | Yes                      | Yes                                  | No                         |
```
Comment (RealOrangeOne) [inline docs/concepts/comparison.rst]: What does this even mean?

File: docs/concepts/comparison.rst [hunk 3]
```diff
@@ -33,19 +31,17 @@ Similar Libraries
...
-| Supports SourceBots Servo Board           | Yes                      | Yes                                  | No                         |
+| Supports SourceBots Servo Board           | Upcoming                 | Yes                                  | No                         |
 +-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 | Supports SR v4 Kit                        | Yes                      | Partial Support                      | Yes                        |
 +-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
-| Supports vendor hardware modules          | Yes                      | No                                   | No                         |
-+-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 | Test Coverage                             | > 97%                    | Some on robot-api                    | No                         |
```
Comment (RealOrangeOne) [inline docs/concepts/comparison.rst]:
```suggestion
| Test Coverage                             | > 97%                    | Some                    | No                         |
```

https://github.com/sourcebots/robotd/blob/master/tests/test_game_state.py

File: docs/concepts/comparison.rst [hunk 4]
```diff
@@ -33,19 +31,17 @@ Similar Libraries
...
 +-------------------------------------------+--------------------------+--------------------------------------+----------------------------+
 | Type Checking                             | Yes                      | Partial                              | No                         |
```
Comment (RealOrangeOne) [inline docs/concepts/comparison.rst]: Type checking isn't available in Python 2
```suggestion
| Type Checking                             | Yes                      | Partial                              | N/A                         |
```

Comment (sedders123) [reply docs/concepts/comparison.rst]: Type annotations aren't available in Python 2. Typing can be achieved through type comments though

File: docs/concepts/comparison.rst (hunk 2 above)
Comment (trickeydan) [reply docs/concepts/comparison.rst]: For example, both `j5` and `robot-api` can support multiple backends, such as a simulated environment. For `robot-api` this requires a separate `robotd` implementation.

File: docs/concepts/comparison.rst (hunk 4 above)
Comment (trickeydan) [reply docs/concepts/comparison.rst]: It is, but uses pre-3.5 syntax: https://mypy.readthedocs.io/en/latest/python2.html

**Pull request merged**